.github/workflows/gcr-cleaner.yml

```
@@ -5,20 +5,29 @@ on:
   schedule:
     - cron: '0 0 */1 * *' # runs daily
   workflow_dispatch: # allows for manual invocation
+env:
+  GCP_WORKLOAD_IDENTITY_PROVIDER: 'projects/765091727073/locations/global/workloadIdentityPools/hato-atama-workload-identity/providers/github'
+  GCP_SERVICE_ACCOUNT: 'gcr-cleaner@hato-atama.iam.gserviceaccount.com'
 
 jobs:
   gcr-cleaner:
     runs-on: 'ubuntu-latest'
+    permissions:
+      id-token: write
+      contents: read
     steps:
-#      # configure based on your registry
-#      - uses: 'docker/login-action@v2'
-#        with:
-#          username: ${{ secrets.DOCKERHUB_USERNAME }}
-#          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - id: 'auth'
+        name: 'Authenticate to GCP'
+        uses: google-github-actions/auth@v1.0.0
+        with:
+          workload_identity_provider: ${{env.GCP_WORKLOAD_IDENTITY_PROVIDER}}
+          service_account: ${{env.GCP_SERVICE_ACCOUNT}}
 
       # customize based on the gcr-cleaner flags
       - uses: 'docker://us-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner-cli'
```

```suggestion
      - uses: 'docker://asia-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner-cli'
```

より日本に近いレジストリを指定してあげても良さそう。
参考: https://github.com/GoogleCloudPlatform/gcr-cleaner#container-images

---

azuru上で実行するactionsのインスタンスが日本にあるとは限らんと思ったのよね。

---

なるほど。